CommandLine/libEGM/CMakeLists.txt

```diff
@@ -36,14 +46,23 @@ target_link_libraries(${LIB} PRIVATE pugixml)
 
 find_package(RapidJSON REQUIRED)
 include_directories(${LIB} PRIVATE ${RAPIDJSON_INCLUDE_DIRS})
-target_link_libraries(${LIB} PRIVATE rapidjson)
+if(MSVC)
+  target_link_libraries(${LIB} PRIVATE rapidjson)
+endif(MSVC)
 
 find_package(yaml-cpp CONFIG REQUIRED)
 target_link_libraries(${LIB} PRIVATE yaml-cpp)
 
 include_directories(. "${ENIGMA_DIR}/CommandLine/protos/" "${ENIGMA_DIR}/shared/lodepng")
-link_directories(${ENIGMA_DIR})
-target_link_libraries(${LIB} PRIVATE Protocols)
+
+find_library(LIB_PROTOCOLS NAMES Protocols PATHS ${ENIGMA_DIR})
+target_link_libraries(${LIB} PRIVATE ${LIB_PROTOCOLS})
+
+#Linux is a turd
+if(NOT MSVC)
+  find_library(LIB_FS NAMES stdc++fs)
+  target_link_libraries(${LIB} PRIVATE ${LIB_FS})
+endif(NOT MSVC)
 
 install(TARGETS ${LIB} DESTINATION "${ENIGMA_DIR}")
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${LIB}.dir/Debug/${LIB}.pdb" DESTINATION "${ENIGMA_DIR}" OPTIONAL)
\ No newline at end of file
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${LIB}.dir/Debug/${LIB}.pdb" DESTINATION "${ENIGMA_DIR}" OPTIONAL)
```

Comment:
Shouldn't this be wrapped in an MSVC check since only that compiler outputs pdb files?